### ISSUE TITLE
Add support for alternate API endpoints

### ISSUE THREAD

**Issue by zzz** [siznax/wptools#127]
Some wikis have different endpoints. It would be helpful to be able to change the default endpoint.

**Comment by seansay** [comment]
hi @zzz 
> Some wikis have different endpoints
Can you give an example of this?  Just for better understanding the issue

**Comment by siznax** [comment]
Thanks for responding, @seansay!

@zzz contacted me about this, and I asked him to open this issue.

It looks like some instances of MediaWiki use different API endpoints, e.g.

English Wikipedia:
https://en.wikipedia.org/w/api.php

Gamepedia:
https://hearthstone.gamepedia.com/api.php

So, `/api.php` instead of `/w/api.php` here. 

I have a patch for this I plan to commit soon...

```python
>>> page = wptools.page('Hearthstone', wiki='https://hearthstone.gamepedia.com', endpoint='/api.php')
>>> page.get_query().get_parse().get_more()
hearthstone.gamepedia.com (query) Hearthstone
hearthstone.gamepedia.com (imageinfo) File:Hearthstone logo.png
hearthstone.gamepedia.com (parse) 8074
hearthstone.gamepedia.com (querymore) 8074
Hearthstone (en) data
{
  WARNINGS: <dict(2)> query, main
  backlinks: <list(352)> {u'ns': 0, u'pageid': 1, u'title': u'Hear...
  contributors: 11
  extext: <str(834)> _**Hearthstone**_ (formerly known as _**Heart...
  extract: <str(884)> <p><i><b>Hearthstone</b></i> (formerly known...
  files: <list(22)> File:Blackrock Mountain.jpg, File:BlizzCon 201...
  image: <list(2)> {u'size': 181505, 'kind': 'query-pageimage', u'...
  iwlinks: <list(6)> https://wow.gamepedia.com/Blizzard_Entertainm...
  length: 16,581
  modified: <dict(1)> page
  pageid: 8074
  parsetree: <str(19838)> <root>[[File:Hearthstone logo.png|center...
  random: Patches the Pirate
  redirects: <list(2)> {u'ns': 0, u'pageid': 43360, u'title': u'Ke...
  requests: <list(4)> query, imageinfo, parse, querymore
  title: Hearthstone
  url: https://hearthstone.gamepedia.com/Hearthstone
  url_raw: https://hearthstone.gamepedia.com/Hearthstone?action=raw
  wikitext: <str(16416)> [[File:Hearthstone logo.png|center]][[Fil...
}
```

**Comment by siznax** [comment]
Hmm... one of our test cases is [failing](https://travis-ci.org/siznax/wptools/jobs/401109410) now, but it seems to be unrelated to recent updates:

```
======================================================================
ERROR: test_request_curl_info (tests.test_basic.WPToolsRequestTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/siznax/wptools/tests/test_basic.py", line 871, in test_request_curl_info
    info = wptools.request.curl_info(req.cobj)
  File "/home/travis/build/siznax/wptools/wptools/request.py", line 125, in curl_info
    "content": crl.getinfo(crl.CONTENT_TYPE),
TypeError: decoding to str: need a bytes-like object, NoneType found
```

That test passes for me in python 3.6.3:

```
$ python --version
Python 3.6.3
$ nosetests tests/test_basic.py:WPToolsRequestTestCase.test_request_curl_info -v
test_request_curl_info (tests.test_basic.WPToolsRequestTestCase) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.004s

OK
```

**Comment by zzz** [comment]
Windows 10, Python 3.6.6.

```
======================================================================
ERROR: test_request_curl_info (tests.test_basic.WPToolsRequestTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Nikolay\Downloads\wptools-master\tests\test_basic.py", line 871, in test_request_curl_info
    info = wptools.request.curl_info(req.cobj)
  File "C:\Users\Nikolay\Downloads\wptools-master\wptools\request.py", line 125, in curl_info
    "content": crl.getinfo(crl.CONTENT_TYPE),
TypeError: decoding to str: need a bytes-like object, NoneType found
```

**Comment by siznax** [comment]
Thanks for the unit test confirmation and for opening this issue @zzz 

Support for this is now available in v0.4.16
